benchmarking/benchmark_framework_module.rb

```diff
@@ -0,0 +1,22 @@
+module BenchmarkFrameworkModule
+  class BenchmarkFramework
+    @@blocks = {}
+    class << self
+      def blocks
+        @@blocks
```

Comment:
There is something to consider when using class variables. Imagine if you're running many benchmarks consecutively and you're constantly mutating a class variable. Would that impact the results? Computers aren't perfect, and memory will not clear entirely every single time so you may be left with some residual data that will make your benches appear faster than what they would be in a fresh state.

I thought this to be an interesting article
http://cranialburnout.blogspot.com/2013/09/avoiding-mutable-state.html

I just did this google search and a lot comes up you can read on your own.
[Why is mutability a bad thing?](https://www.google.com/search?q=why+is+mutability+a+bad+thing&oq=why+is+mutability+a+bad+thing&aqs=chrome..69i57.5827j0j7&sourceid=chrome&ie=UTF-8)

You can't avoid mutability entirely but a good way of making is safer is by using instances of objects along with instance variables rather than using the object of a class itself to maintain state.